#! /bin/sh /usr/share/dpatch/dpatch-run
## sqlite3-support.dpatch by Diego Iastrubni <diego.iastrubni@xorcom.com>
##
## DP: SQLite3 support. See upstream http://www.freepbx.org/trac/ticket/1784

@DPATCH@
diff -urNad freepbx-module-disa~/install.php freepbx-module-disa/install.php
--- freepbx-module-disa~/install.php	2006-05-31 13:16:43.000000000 +0300
+++ freepbx-module-disa/install.php	2007-02-18 13:23:32.000000000 +0200
@@ -1,12 +1,41 @@
 <?php
 
 global $db;
+global $amp_conf;
+
+$autoincrement = (($amp_conf["AMPDBENGINE"] == "sqlite") || ($amp_conf["AMPDBENGINE"] == "sqlite3")) ? "AUTOINCREMENT":"AUTO_INCREMENT";
+
+$sql = "CREATE TABLE IF NOT EXISTS manager (
+	`manager_id` INTEGER NOT NULL  PRIMARY KEY $autoincrement,
+	`name` VARCHAR( 15 ) NOT NULL ,
+	`secret` VARCHAR( 50 ) ,
+	`deny` VARCHAR( 255 ) ,
+	`permit` VARCHAR( 255 ) ,
+	`read` VARCHAR( 50 ) ,
+	`write` VARCHAR( 50 )
+)";
+
+$sql = "CREATE TABLE IF NOT EXISTS disa ( 
+	disa_id INTEGER NOT NULL PRIMARY KEY $autoincrement,
+	displayname VARCHAR( 50 ), 
+	pin VARCHAR ( 50 ), 
+	cid VARCHAR ( 50 ), 
+	context VARCHAR ( 50 ), 
+	digittimeout INTEGER, 
+	resptimeout INTEGER, 
+	needconf VARCHAR( 10 ) 
+);";
+
+$check = $db->query($sql);
+if (DB::IsError($check)) {
+	die( "Can not create `disa` table: " . $check->getMessage() .  "\n");
+}
 
 // Manage upgrade from DISA 1.0
 // r2.0 Add Timeouts and add wait for confirmation
 $sql = "SELECT digittimeout FROM disa";
 $check = $db->getRow($sql, DB_FETCHMODE_ASSOC);
-if(DB::IsError($check)) {
+if (DB::IsError($check)) {
 	// add new fields - Digit Timeout
 	$sql = 'ALTER TABLE disa ADD COLUMN digittimeout INT DEFAULT "5"';
 	$result = $db->query($sql);
@@ -26,5 +55,4 @@
 	}
 }
 
-
 ?>
diff -urNad freepbx-module-disa~/install.sql freepbx-module-disa/install.sql
--- freepbx-module-disa~/install.sql	2006-05-31 13:16:43.000000000 +0300
+++ freepbx-module-disa/install.sql	1970-01-01 02:00:00.000000000 +0200
@@ -1 +0,0 @@
-CREATE TABLE IF NOT EXISTS disa ( disa_id INT NOT NULL AUTO_INCREMENT PRIMARY KEY , displayname VARCHAR( 50 ), pin VARCHAR ( 50 ), cid VARCHAR ( 50 ), context VARCHAR ( 50 ), digittimeout INT, resptimeout INT, needconf VARCHAR( 10 ) ); 
diff -urNad freepbx-module-disa~/uninstall.sql freepbx-module-disa/uninstall.sql
--- freepbx-module-disa~/uninstall.sql	2006-05-31 13:16:43.000000000 +0300
+++ freepbx-module-disa/uninstall.sql	2007-02-18 13:23:32.000000000 +0200
@@ -1 +1 @@
-DROP TABLE disa;
+DROP TABLE IF EXISTS disa;
